lib/msprime.c

```diff
@@ -1263,21 +1263,32 @@ msp_defrag_segment_chain(msp_t *self, segment_t *z)
     return 0;
 }
 
+static int64_t
+msp_dtwf_generate_breakpoint(msp_t *self, int64_t start)
+{
+    double mu = 1.0 / self->recombination_rate;
+    double x = gsl_ran_exponential(self->rng, mu);
+
+    /* Prevent overflowing into a negative number when we convert to int
+     * and clamp to UINT32_MAX since we can't have anything bigger than
+     * this anyway. */
+    x = GSL_MIN(UINT32_MAX, x);
+    assert(x >= 0);
```

Comment:
Any benefit to just setting `x = UINT32_MAX` if `x < 0`? We'd avoid having an assertion then, but maybe that's masking too much.

Reply:
I don't think x can be negative at this point, as gsl_ran_exponential should always return something positive. We only got negative values because we cast to an int, and the value was too large for an int64, causing it to flip negative. But, just to be sure, I stuck in an assert!